pkg/controller/operators/catalog/operator.go

```diff
@@ -881,7 +881,21 @@ func (o *Operator) ensureInstallPlan(logger *logrus.Entry, namespace string, sub
 	for _, installPlan := range installPlans {
 		if installPlan.Status.CSVManifestsMatch(steps) {
 			logger.Infof("found InstallPlan with matching manifests: %s", installPlan.GetName())
-			return reference.GetReference(installPlan)
+
+			for _, sub := range subs {
+				ownerutil.EnsureOwner(installPlan, sub)
+			}
+			updatedPlan, err := o.client.OperatorsV1alpha1().InstallPlans(installPlan.GetNamespace()).Update(installPlan)
```

Comment:
shouldn't we only attempt to update if the owners have changed?

Reply:
Sure, updated.